File: fis-nutricion-infantil/backend/src/main/java/backend/service/TutorService.java
package backend.service;

import backend.model.Tutor;
import backend.repository.TutorRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class TutorService {
    private final TutorRepository tutorRepository;

    public TutorService(TutorRepository tutorRepository) {
        this.tutorRepository = tutorRepository;
    }

    public Tutor guardar(Tutor tutor) {
        return tutorRepository.save(tutor);
    }

    public List<Tutor> listar() {
        return tutorRepository.findAll();
    }

    public Optional<Tutor> buscarPorId(Integer id) {
        return tutorRepository.findById(id);
    }

    public void eliminar(Integer id) {
        tutorRepository.deleteById(id);
    }
}

File: fis-nutricion-infantil/frontend/src/services/tutorService.js
import apiClient from './apiClient'

export const tutorService = {
  listar: () =>
    apiClient.get('/tutores').then((response) => response.data),

  buscarPorId: (id) =>
    apiClient.get(`/tutores/${id}`).then((response) => response.data),

  guardar: (data) =>
    apiClient.post('/tutores', data).then((response) => response.data),

  actualizar: (id, data) =>
    apiClient.put(`/tutores/${id}`, data).then((response) => response.data),

  eliminar: (id) =>
    apiClient.delete(`/tutores/${id}`).then((response) => response.data),
}